bittensor/utils/__init__.py

```diff
@@ -63,15 +63,19 @@ def unbiased_topk( values, k, dim=0, sorted = True, largest = True):
     return topk, permutation[ indices ]
 
 
-def version_checking():
-    response = requests.get(bittensor.__pipaddress__)
-    latest_version = response.json()['info']['version']
-    version_split = latest_version.split(".")
-    latest_version_as_int = (100 * int(version_split[0])) + (10 * int(version_split[1])) + (1 * int(version_split[2]))
+def version_checking( timeout : int = 15):
+    try:
+        response = requests.get(bittensor.__pipaddress__, timeout=timeout)
+        latest_version = response.json()['info']['version']
+        version_split = latest_version.split(".")
+        latest_version_as_int = (100 * int(version_split[0])) + (10 * int(version_split[1])) + (1 * int(version_split[2]))
 
-    if latest_version_as_int > bittensor.__version_as_int__:
-        print('\u001b[33mBittensor Version: Current {}/Latest {}\nPlease update to the latest version at your earliest convenience\u001b[0m'.format(bittensor.__version__,latest_version))
+        if latest_version_as_int > bittensor.__version_as_int__:
+            print('\u001b[33mBittensor Version: Current {}/Latest {}\nPlease update to the latest version at your earliest convenience\u001b[0m'.format(bittensor.__version__,latest_version))
+    except Exception:
+        pass
```

Comment:
It would be good to specify either what an exception would be or put down an error message here.

Reply:
@smckay42 something like:

```python
def version_checking( timeout : int = 15):
    try:
        response = requests.get(bittensor.__pipaddress__, timeout=timeout)
        latest_version = response.json()['info']['version']
        version_split = latest_version.split(".")
        latest_version_as_int = (100 * int(version_split[0])) + (10 * int(version_split[1])) + (1 * int(version_split[2]))

        if latest_version_as_int > bittensor.__version_as_int__:
            print('\u001b[33mBittensor Version: Current {}/Latest {}\nPlease update to the latest version at your earliest convenience\u001b[0m'.format(bittensor.__version__,latest_version))
    
    except requests.exceptions.Timeout:
        bittensor.logging.error('Version check failed due to timeout')
    except requests.exceptions.RequestException as e:
        bittensor.logging.error(f'Version check failed due to request failure: {e}')
```